editor/editor_help_search.cpp

```diff
@@ -590,16 +657,43 @@ TreeItem *EditorHelpSearch::Runner::_create_class_hierarchy(const ClassMatch &p_
 	return class_item;
 }
 
+TreeItem *EditorHelpSearch::Runner::_find_or_create_item(TreeItem *p_parent, const String &p_item_meta) {
+	TreeItem *item = nullptr;
+
+	// Attempt to find in cache.
+	if (tree_cache->item_cache.has(p_item_meta)) {
+		item = tree_cache->item_cache[p_item_meta];
+
+		// Remove from cache.
+		tree_cache->item_cache.erase(p_item_meta);
+
+		// Add to tree.
+		p_parent->add_child(item);
+
+		// Erase buttons if any, to update icons.
```

Comment:
This part isn't optimal perhaps, and unsure if the items would even change this between runs, but did it here to avoid having to alter the behaviour based on if the item was fetched or not